spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/DefaultAmqpHeaderMapper.java

```diff
@@ -359,7 +360,8 @@ private String extractContentTypeAsString(Map<String, Object> headers) {
 		if (contentType != null) {
 			String contentTypeClassName = contentType.getClass().getName();
 
-			if (contentTypeClassName.equals("org.springframework.http.MediaType")) { // see INT-2713
+			if (contentType instanceof MimeType ||
```

Comment:
Don't see reason to worry about classloader issue, because `MimeType` is a part of spring-core, as you realized.

If I understand you correctly you want to break it a bit and do `contentType.toString()` independently of its type.
I won't mind, but you is right: it is a breaking change and must be done only for the 4.2. So, new JIRA?

Any objection to merge and cherry-pick this as is?